x/delayedack/keeper/hooks.go

```diff
@@ -28,3 +35,40 @@ func (k eibcHooks) AfterDemandOrderFulfilled(ctx sdk.Context, demandOrder *eibct
 	}
 	return nil
 }
+
+/* -------------------------------------------------------------------------- */
+/*                                 epoch hooks                                */
+/* -------------------------------------------------------------------------- */
+var _ epochstypes.EpochHooks = epochHooks{}
+
+type epochHooks struct {
+	Keeper
+}
+
+func (k Keeper) GetEpochHooks() epochstypes.EpochHooks {
+	return epochHooks{
+		Keeper: k,
+	}
+}
+
+// BeforeEpochStart is the epoch start hook.
+func (e epochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook.
+// We want to clean up the demand orders that are with underlying packet status which are finalized.
+func (e epochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	if epochIdentifier != e.GetParams(ctx).EpochIdentifier {
+		return nil
+	}
+	// Get all rollapp packets with status FINALIZED and delete them
+	finalizedRollappPackets := e.ListRollappPacketsByStatus(ctx, commontypes.Status_FINALIZED, 0)
+	for _, finalizedPacket := range finalizedRollappPackets {
+		finalizedPacketCopy := finalizedPacket
+		_ = osmoutils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
```

Comment:
log errors?

Reply:
from the godoc of the `ApplyFuncIfNoError`:

```
// This function lets you run the function f, but if theres an error or panic
// drop the state machine change and log the error.
```